### PR TITLE
TEC-4016 Move all editor compatibility into its own file(s). Update Classic Editor compatibilty.

### PR DESCRIPTION
**Pull request by Camwyn**
This PR implements a namespace for the Editor and creates a Compatibility section in it.

Functions that are solely for the "Classic Editor" plugin are moved to that section, out of the main Editor functionality.
All third-party compatibility functionality is now implemented via hooks.

Deprecates unused functions and those moved.

[TEC-4016]

[TEC-4016]: https://theeventscalendar.atlassian.net/browse/TEC-4016

### CODE REVIEW

File: src/Tribe/Editor.php [hunk 1]
```diff
@@ -8,7 +8,16 @@
...
 
 	/**
-	 * Meta key for flaging if a post is from classic editor
+	 * Key we store the toggle under in the tribe_events_calendar_options array.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $blocks_editor_key = 'toggle_blocks_editor';
```
Comment (bordoni) [inline src/Tribe/Editor.php:17]: use `static`

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tribe/Editor/Compatibility/Classic_Editor.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	}
+
+	public function hooks() {
+		add_filter( 'tribe_editor_should_load_blocks', [ $this, 'filter_tribe_editor_should_load_blocks' ] );
+	}
+
+	public function filter_tribe_editor_should_load_blocks( $should_load_blocks ) {
+		global $post;
+
+		if ( ! self::is_classic_plugin_active() ) {
```
[the remark this answers is not in the document]
Comment (bordoni) [reply src/Tribe/Editor/Compatibility/Classic_Editor.php:124]: hahaha

File: src/Tribe/Editor.php [hunk 3]
```diff
@@ -60,49 +66,26 @@ public function is_wp_version() {
...
+	 * Checks if we have the Gutenberg Project plugin active.
 	 *
 	 * @since 4.8
 	 *
 	 * @return boolean
 	 */
-	public function is_gutenberg_active() {
-		return function_exists( 'the_gutenberg_project' );
+	private function is_gutenberg_plugin_active() {
+		return function_exists( 'gutenberg_register_scripts_and_styles' );
```
[the remark this answers is not in the document]
Comment (Camwyn) [reply src/Tribe/Editor.php]: 🤦🏼

File: src/Tribe/Editor.php [hunk 4]
```diff
@@ -60,49 +67,26 @@ public function is_wp_version() {
...
-	 * its a external plugin
+	 * Checks if we have the Gutenberg Project plugin active.
 	 *
 	 * @since 4.8
 	 *
 	 * @return boolean
 	 */
-	public function is_gutenberg_active() {
-		return function_exists( 'the_gutenberg_project' );
+	private function is_gutenberg_active() {
```
Comment (borkweb) [inline src/Tribe/Editor.php:76]: We can't change the visibility of this function without risking breakages for folks.
```suggestion
	public function is_gutenberg_active() {
```

**Review by borkweb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**